### ISSUE TITLE
Regarding the noise models

### ISSUE THREAD

**Issue by karnikram** [kissb2/PyICP-SLAM#2]
Hi,

Thank you for this project, it has been very useful.

I was wondering if there is any heuristic for choosing the covariance values that you have used here, and to what extent they affect the optimization.

```python
self.prior_cov = minisam.DiagonalLoss.Sigmas(np.array([1e-6, 1e-6, 1e-6, 1e-4, 1e-4, 1e-4])) 
self.const_cov = np.array([0.5, 0.5, 0.5, 0.1, 0.1, 0.1])
self.odom_cov = minisam.DiagonalLoss.Sigmas(self.const_cov)
self.loop_cov = minisam.DiagonalLoss.Sigmas(self.const_cov)
```
Any insight on this would be helpful.

**Comment by gisbi-kim** [comment]
Thank you for your interest in this repo.

As you said, I think the values are generally selected heuristically.

First, for the prior, to prevent "gauge problem", usually very small covariance values are used. 
and for the others, from my experiences, not this repo, I empirically found the adequate value via watching the quality of trajectory and the optimized map. 
At that time, I used a GPS factor and used its covariance as the sensor gave (or the number of satellites), but in a long straight flat road, the optimized trajectory fluctuated vertically so I just used a fixed constant value for all places and equal value for latitude, longitude, and altitude. 

So I think it's up to you in a real-world project and dependent on the environment's characteristics.
